client/src/partials/cash/modals/transfer.modal.js

```diff
@@ -0,0 +1,113 @@
+angular.module('bhima.controllers')
+.controller('CashTransferModalController', CashTransferModalController);
+
+CashTransferModalController.$inject = [
+  '$uibModalInstance', '$q', 'SessionService', 'CurrencyService', 'VoucherService', 'CashboxService', 'AccountService', 'cashBox', 'uuid'
+];
+
+/**
+ * @module cash/modals/CashTransferModalController
+ *
+ * @description This controller is responsible transfering money between auxillary cash and a virement account
+*/
+
+function CashTransferModalController(ModalInstance, $q, sessionService, currencyService, voucherService, cashBoxService, accountService, cashBox, uuid) {
+  var vm = this; 
+
+  /** Attaching service to the scope **/
+  vm.cashBox = cashBox;
+  vm.currencyService = currencyService; 
+  vm.cashBoxService = cashBoxService;
+
+  /** check if the cashbox id is provided**/
+  vm.cashBoxIdMissed = !cashBox.id;
+
+  /**
+  * @function submit
+  * @param {boolean} invalid it is true if the form is valid
+  * @description for submitting a dialog content
+  **/
+  function submit (invalid){
+    if (invalid) { return; }
+    var voucher = processVoucher();
```

Comment:
One of the goals of bhima-2.X was to remove this kind of preprocessing from the `controller`.  It should be done either 
 1. in a service or
 2. on the server.

I would prefer it was done on the server, since we can easily write many integration tests for corner cases.  Could you move this code either to a service or the server?

If you choose to move this code onto the server, the API should accept a POST request with a body like this:
```js
{ 
  uuid : 'some uuid',   /* this could also be empty! */
  currency_id : 1,

  /* ...  other voucher properties ... */

 items : [{
    uuid : 'some uuid',  /* this could also be empty! */
    debit : 12.2,
    credit : 0
  }, {
    uuid : 'some uuid',  /* this could also be empty! */
    debit : 0,
    credit : 12.2
  }]
}
```
This is similar to how the Cash API works.  The server will then take these items and process them into an array of arrays for insertion into the database.  See [the cash API code](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/server/controllers/finance/cash.js#L171) for more details.